python/graphstorm/config/argument.py

```diff
@@ -614,11 +615,14 @@ def model_encoder_type(self):
         """ Which graph encoder to use, it can be GNN or language model only
         """
         # pylint: disable=no-member
-        assert hasattr(self, "_model_encoder_type"), \
-            "Model encoder type should be provided"
-        assert self._model_encoder_type in BUILTIN_ENCODER, \
-            f"Model encoder type should be in {BUILTIN_ENCODER}"
-        return self._model_encoder_type
+        if "distill" not in self.configuration["gsf"]:
```

Comment:
Why we need this?
What is the encoder type for distillation? Should it be LM? or you can add a new one called distill_lm.

Reply:
there'll be no encoder type for distillation. However, for non-distillation task, this is a mush-have argument, so we need to have a check here.

Reply:
shouldn't the encoder be BERT for distillation?

Reply:
encoder_type is defined under "gsf -> gnn" section in the yaml config. I feel it's better to restrict its scope by "the encoder for node features for graph related training". However, distillation has nothing to do with the graph, so it confuses people if using the same encoder type argument here.

Reply:
We have self.distill_lm_configs now. We can do:
```
If self.distill_lm_configs is not None:
   assert xxx
else:
  return None
```